CMakeLists.txt

```diff
@@ -277,6 +277,12 @@ if("${is_system_dir}" STREQUAL "-1")
 set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/${LIB_INSTALL_DIR}")
 endif()
 
+#Used in Simulation / simulation_io
+find_package(GLEW)
```

Comment:
I'd recommend against this, since `BUILD_simulation=OFF` should not require these packages

Reply:
But they aren't required even if it is off. They are indeed search for and maybe found, but not necessarily used.

If simulation is ON, they will be linked against, but if they are not found - it will atm silently just not build / add the projects.

All I/O unit frameworks are search for, but I guess its rarely that you have them all installed?

Reply:
I have updated it to be guarded with WITH_GLEW/GLUT.

Reply:
> All I/O unit frameworks are search for

I didn't see that. I'm not much of a CMake wizard. Maybe @taketwo @SergioRAgostinho can pip in with what's actually recommended. *[kunaltyagi] exists stage*

Reply:
I don't consider the state of our CMake files to be pristine, hence I would say to rely on old behaviors with some care. It is true that all grabber 3rd partiy deps are checked irrespective of them being built, but I would not consider that to be the ideal case.

Hence in my ideal world, I would only search for GLEW and GLUT if needed. If trying to achieve that is hugely laborious then I would consider the current approach you took as a necessary compromise.